rfcs/icon-component.md

```diff
@@ -0,0 +1,69 @@
+# Icon Component
+
+This is the set of ~16 icons used internally within baseui that are safe to open source. This is not the set of uber-specific icons published internally as @uber/icons (though the API is intended to align with that to reduce confusion).
+
+![image](https://user-images.githubusercontent.com/875591/45131331-51e8c500-b141-11e8-898d-ead83d44b345.png)
+
+## Usage
+
+```javascript
+import * as React from 'react';
+import {LeftArrow} from 'baseui/icon/left-arrow';
+
+export default () => (
+  <LeftArrow size="10px" color="#ccc" />
+);
+```
+
+## Exports
+
+* `Icon` (Base Icon component used by other exports below)
+
+From left-to-right based on the image above:
+
+* `DeleteAlt`
+* `Search`
+* `Delete`
+* `Check`
+* `ArrowUp`
+* `ArrowDown`
+* `ArrowLeft`
+* `ArrowRight`
+* `TriangleRight`
+* `TriangleDown`
+* `Upload`
+* `Plus`
+* `Grab`
+* `Alert`
+* `Menu`
+* `ChevronRight`
+* `ChevronLeft`
+* `Spinner`
+* `Overflow`
+* `TriangleLeft`
+* `TriangleUp`
+* `Filter`
+* `CheckIndeterminate`
+
+## `Icon` API
+
+* `size: number|string` - Optional, Defaults to `scale500` (12px)
+  Size of element, will be passed to the svg width/height style. Can also be a baseui `theme.sizing` constant like `sizing500`.
+* `color: string` - Optional, Defaults to `currentColor`
+  Color of icon, will be used as svg fill
+* `title: string` - Optional
+  Allows you to override the SVG `<title>` label, which is used for accessibility
+* `overrides: {Svg}`
```

Comment:
What's the use-case of overriding the SVG with another SVG?

Reply:
It's less to override the code, and more for overriding styles or props:

```js
<LeftArrow overrides={{
  Svg: {
    props: {viewBox: '0 0 24 20'},
    style: {position: 'absolute', top: '0px', left: '10px'}
  }
/>
```

Reply:
I see! So this only overrides the top Element's props I suppose?

It's probably clearer if users create a container to house the SVG themselves. I don't think touching the viewBox is a good idea, users might want to touch `preserveAspectRatio` though. 

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio